### PR TITLE
Calendar, Categories, and End Dates

### PR DESCRIPTION
**Pull request by Gabe-Alonso**
Adds Calendar input for dateClosed. dateClosed and Category correctly appear in prompts in the feed

### CODE REVIEW

**Comment by 1colinmcdonald** [comment]
It's not passing the checks. I'm not the expert on the CI/CD though.

**Review by 1colinmcdonald — CHANGES_REQUESTED** [review]
I think it would be a good idea for it to pass the CI/CD checks.

**Review by EmuMan — APPROVED** [review]
Fixed the issue with dependencies, all tests pass now.

**Review by Thundderr — APPROVED** [review]
Sorry but my latest merge with #45 created a conflict with Prompt.js. I'll resolve it rq but otherwise it looks good to go!

**Comment by Thundderr** [comment]
Merge resolved and deploys 👍

**Review by 1colinmcdonald — APPROVED** [review]
Looks good

File: .vscode/settings.json [hunk 1]
```diff
@@ -0,0 +1,2 @@
+{
```
Comment (svenlinus) [inline .vscode/settings.json:1]: What is this file? If it is empty it should not be included in the pr. One could add it to the .gitignore

File: packages/express-backend/services.js [hunk 2]
```diff
@@ -171,7 +171,9 @@ function deleteBetById(id) {
 // PROMPTS
 
 function addPrompt(prompt) {
+  console.log(prompt);
```
Comment (svenlinus) [inline packages/express-backend/services.js]: Are these debug statements still necessary.

File: packages/react-frontend/src/components/Bet.js [hunk 3]
```diff
@@ -14,30 +14,43 @@ import {
   Stack,
 } from "@mui/material";
 
-function Bet(prop) {
+function Bet(props) {
```
Comment (svenlinus) [inline packages/react-frontend/src/components/Bet.js:17]: Nice

File: packages/react-frontend/src/components/Prompt.js [hunk 4]
```diff
@@ -164,47 +175,47 @@ const Prompt = ({
...
-            <Button variant="contained" color="primary" onClick={() => handleBetSubmission(true)}>Yes!</Button>
-            <Button variant="contained" color="secondary" onClick={() => handleBetSubmission(false)}>No!</Button>
-          </Stack>
-        </>
-        :
-        <Typography variant="body1" style={{ textAlign: 'center', marginTop: '10px' }}>
-          You've already bet on this prompt.
-        </Typography>
+          <Typography variant="body1" style={{ textAlign: 'center', marginTop: '10px' }}>
+            You cannot vote on your own prompt.
```
Comment (svenlinus) [inline packages/react-frontend/src/components/Prompt.js:147]: Vote is not the right word here. Maybe bet?

File: packages/react-frontend/src/components/QuestionForm.js [hunk 5]
```diff
@@ -0,0 +1,85 @@
+import React, { useState } from "react";
+import { Box, Typography, Stack, Button, TextField } from "@mui/material";
+import { Prompt } from "./Prompt.js";
+import Feed from "./Feed.js";
+
+function QuestionForm(props){
+  
+
+  const [question, setQuestion] = useState({
+    question: ""
```
Comment (svenlinus) [inline packages/react-frontend/src/components/QuestionForm.js:10]: There is no need for these hooks to be objects, they could all be string.
i.e.
const [question, setQuestion] = useState("");

File: packages/react-frontend/src/components/QuestionForm.js [hunk 6]
```diff
@@ -0,0 +1,85 @@
...
+
+  const [dateClosed, setDateClosed] = useState({
+    dateClosed: ""
+  });
+
+  function handleChange(event) {
+    console.log("we assigning the new stuff");
+    const { name, value } = event.target;
+    if (name === "question")
+        setQuestion({ question: value });
```
Comment (svenlinus) [inline packages/react-frontend/src/components/QuestionForm.js:25]: And then
setQuestion(value);

File: packages/react-frontend/src/components/QuestionForm.js [hunk 7]
```diff
@@ -0,0 +1,85 @@
...
+    
+  }
+
+  function submitForm() {
+    
+    const subPrompt = {
+        question: question.question,
+        category: category.category,
+        dateClosed: dateClosed.dateClosed
+    }
```
Comment (svenlinus) [inline packages/react-frontend/src/components/QuestionForm.js:39]: And then
{
question: question,
category: categort
...
}

**Pull request merged**